lib/puppet/provider/common.rb

```diff
@@ -121,3 +121,8 @@ def extract_resource_name(class_name)
   shift_prefix = Regexp.last_match.nil? ? 2 : 1
   class_name.split('::')[2].split('_').drop(shift_prefix).collect(&:capitalize).join
 end
+
+def create_image_streamer_client
```

Comment:
How about:
```ruby
def self.create_image_streamer_client
  <do what you gotta do>
end

def create_image_streamer_client
  self.class.create_image_streamer_client
end
```
Also, since it is really small maybe just add it to the `self.client` and use this syntax on the `client`, that way we dish out these methods and avoid duplication in one go.